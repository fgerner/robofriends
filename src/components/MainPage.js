import React, {Component} from "react";
import CardList from "../components/CardList";
import SearchBox from '../components/SearchBox'
import Scroll from '../components/Scroll';
import './MainPage.css';
import {robots} from "../robots";

class MainPage extends Component {

    componentDidMount() {
        this.props.onRequestRobots();
    }

    filterRobots = () => {
        return this.props.robots.filter(robot => {
            return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase())
        })
    }

    render() {
        const {onSearchChange, robots, isPending} = this.props;

        return isPending ?
            <h1 className={'tc f1'}>Loading...</h1>
            :
            (
                <div className={'tc'}>
                    <h1 className={'f1'}>RoboFriends</h1>
                    <SearchBox searchChange={onSearchChange}/>
                    <Scroll>
                        <CardList robots={this.filterRobots()}/>
                    </Scroll>
                </div>
            )
    }
}

export default MainPage;