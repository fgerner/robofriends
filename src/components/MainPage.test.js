import {shallow} from 'enzyme';
import MainPage from './MainPage';

let wrapper;

beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false
    }
    wrapper = shallow(<MainPage {...mockProps}/>);
});

describe('Mainpage component', () => {
    it('should renders without crashing', function () {
        expect(wrapper).toMatchSnapshot();
    });

    it('should filter robots correctly', function () {
        const newMockProps = {
            onRequestRobots: jest.fn(),
            robots: [{
                id: 3,
                name: 'bob',
                email: 'test@test.com'
            }],
            searchField: 'a'
        }
        wrapper = shallow(<MainPage {...newMockProps}/>);
        expect(wrapper.instance().filterRobots()).toEqual([])
    });

    xit('should display pending wile loading', function () {
        const mockProps = {
            isPending: true
        }

    });

})

