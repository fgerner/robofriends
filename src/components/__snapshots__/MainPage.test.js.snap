// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mainpage component should renders without crashing 1`] = `ShallowWrapper {}`;
